src/main/java/darwin/parser/TaskParser.java

```diff
@@ -0,0 +1,37 @@
+package darwin.parser;
+
+import darwin.exception.IllegalTaskArgumentException;
+import darwin.exception.IllegalTaskTypeException;
+import darwin.task.Task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * TaskParser interface to parse task arguments into Task objects.
+ */
+public interface TaskParser {
+    Task parse(String taskArgs) throws IllegalTaskArgumentException;
+    Task parseFromDb(String taskStr) throws IllegalTaskTypeException, IllegalTaskArgumentException;
+
+    default void handleTaskStatus(Task task, String taskStatus) throws IllegalTaskArgumentException {
+        switch (taskStatus) {
+        case "1" -> task.markDone();
+        case "0" -> task.unmarkDone();
+        default -> throw new IllegalTaskArgumentException("");
+        }
```

Comment:
This doesn't follow the switch statement form shown in the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#layout):

```java
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```